RFC-007-SWAP-Ethereum-basic.md

```diff
@@ -0,0 +1,315 @@
+    - [Refund](#refund)
+- [Registry extension](#registry-extension)
+- [Examples](#examples)
+    - [RFC003 SWAP REQUEST](#rfc003-swap-request)
+    - [RFC003 SWAP RESPONSE](#rfc003-swap-response)
+    - [HTLC](#htlc)
+
+<!-- tocstop -->
+
+## Description
+
+This RFC defines how to execute a [RFC003](./RFC-003-SWAP-basic.md) SWAP where one of the ledgers is Ethereum and the associated asset is the native Ether asset.
+
+For definitions of the Ethereum ledger and Ether asset see [RFC006](./RFC-004-SWAP-Ethereum.md).
+
+To fulfil the requirements of [RFC003](./RFC-003-SWAP-basic.md) this RFC defines:
+
+- The [identity](./RFC-003-SWAP-basic.md#identity) to be used when negotiating a SWAP on the Ethereum ledger.
+- How to construct a Hash Time Lock Contract (HTLC) to lock the Ether asset on the Ethereum blockchain.
+- How to deploy, redeem and refund the HTLC during the execution phase of [RFC003](./RFC-003-SWAP-basic.md).
+
+## The Ethereum Identity
+
+The Identity to be used on Ethereum is the Ethereum *address* as defined in section 284 of the [Ethereum Yellowpaper](https://ethereum.github.io/yellowpaper/paper.pdf).
+It is defined as the right most 160-bits (20 bytes) of the Keccak-256 hash of the corresponding ECDSA public key.
+
+In the JSON encoding, an ethereum address MUST be encoded as this 20 byte hex string prefixed by `0x` (as is standard in the Ethereum ecosystem).
+Furthermore, implementations MUST also accept [EIP50](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-55.md) mixed case addresses and MAY verify the checksum.
+
+This RFC extends the [registry](./registry.md) with the following entry in the identity table:
+
+| Ledger   | Identity Name | JSON Encoding         | Description         |
+|:---------|:--------------|:----------------------|---------------------|
+| Ethereum | `address`     | `0x` prefixed address | An Ethereum Address |
+
+Note that since *contract* addresses and *user* addresses are indistinguishable, technically the address used in a SWAP could be that of a contract rather than a user.
+RFCs that use the Ethereum Identity defined here should explain the impact (if any) this has on the protocol.
+There is nothing inherently wrong from the perspective of this RFC about using a contract address as an identity.
+
+## Hash Time Lock Contract
+
+### Hash Functions
+
+This RFC specifies SHA-256 as the only value the `hash_function` parameter to `comit-rfc-003` may take if Ethereum is used as a ledger.
+This may be expanded in subsequent RFCs.
+
+### Parameters
+
+The parameters for the Ether HTLC follow [RFC003](./RFC-003-SWAP-basic.md#hash-time-lock-contract-htlc) and are described concretely in the following table:
+
+| Variable        | Description                                                                 |
+|:----------------|:----------------------------------------------------------------------------|
+| asset           | The quantity of wei                                                         |
```

Comment:
```suggestion
| `asset`           | The quantity of wei                                                         |
```

All parameters are back-ticked in the other draft so I'd expect consistency.

Reply:
This is a different table with different semantics.